build-support/scripts/protobuf.sh

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/env bash
+
+readonly SCRIPT_NAME="$(basename ${BASH_SOURCE[0]})"
+readonly SCRIPT_DIR="$(dirname "${BASH_SOURCE[0]}")"
+readonly SOURCE_DIR="$(dirname "$(dirname "${SCRIPT_DIR}")")"
+readonly FN_DIR="$(dirname "${SCRIPT_DIR}")/functions"
+
+source "${SCRIPT_DIR}/functions.sh"
+
+unset CDPATH
+
+set -euo pipefail
+
+usage() {
+cat <<-EOF
+Usage: ${SCRIPT_NAME} [<options ...>]
+
+Description:
+    Generate all Go files from protobuf definitions. In addition to
+    running the protoc generator it will also fixup build tags in the
+    generated code and regenerate mog outputs and RPC stubs.
+
+Options:
+    --protoc-bin             Path to protoc.
+    --tools-only             Install all required tools but do not generate outputs.
+    -h | --help              Print this help text.
+EOF
+}
+
+err_usage() {
+    err "$1"
+    err ""
+    err "$(usage)"
+}
+
+main() {
+    local protoc_version=
+    local tools_only=
+
+    while test $# -gt 0
+    do
+        case "$1" in
+            -h | --help )
+                usage
+                return 0
+                ;;
+            --protoc-version )
+                protoc_version="$2"
+                shift 2
+                ;;
+            --tools-only )
+                tools_only=1
+                shift
+                ;;
+        esac
+    done
+
+    if test -z "${protoc_version}"
+    then
+        protoc_version="$(make --no-print-directory print-PROTOC_VERSION)"
```

Comment:
My mystery stdout/stderr output was actually coming from how `make` changes behavior when it detects it's been invoked by another copy of `make`. The inner `make` prints stuff when this happens, which was messing up my `make print-ENV_VAR_NAME` trick.

`make --no-print-directory TARGET` disables that "feature".

Reply:
This trick is nice because it keeps these random project version numbers up at the toplevel in the makefile and not buried down inside of some random helper bash script.